src/iris/api.py

```diff
@@ -3900,6 +3922,52 @@ def on_put(self, req, resp, username):
         resp.status = HTTP_204
 
 
+class UserTemplateOverrides(object):
+    allow_read_no_auth = False
+    enforce_user = True
+
+    def on_post(self, req, resp, username):
+        # check request body integrity
+        try:
+            data = ujson.loads(req.context['body'])
+        except ValueError:
+            raise HTTPBadRequest('Invalid json in post body')
+        if not isinstance(data.get('template_overrides'), dict):
+            raise HTTPBadRequest('Invalid json in post body')
+
+        # currently only sms is needed/supported
+        override_val = data.get('template_overrides', {}).get('sms')
+        if not override_val:
+            raise HTTPBadRequest('No valid mode override values in post body')
+
+        # update mode template override settings for user
+        connection = db.engine.raw_connection()
+        cursor = connection.cursor()
+
+        if override_val == 'enabled':
+            try:
+                cursor.execute(insert_user_modes_template_override_query, req.context['username'])
+                connection.commit()
+            except Exception:
+                logger.exception('Failed setting mode template override for user %s', req.context['username'])
+        elif override_val == 'disabled':
+            try:
+                cursor.execute(delete_user_modes_template_override_query, req.context['username'])
+                connection.commit()
+            except Exception:
+                logger.exception('Failed setting mode template override for user %s', req.context['username'])
+        else:
+            cursor.close()
+            connection.close()
+            raise HTTPBadRequest('Invalid sms override setting value')
```

Comment:
Maybe also print what the value was? 

`raise HTTPBadRequest(f'Invalid sms override setting value: {override_val}')`

Reply:
sure, changed